### PR TITLE
feat: merge connector and operator repos into this repo

### PR DESCRIPTION
**Pull request by donch1989**
Because
- As our project expands to include various component types such as generic, iterator, and AI components, managing separate repositories for each becomes less practical. Although the initial plan aimed to integrate them directly into the component repository, the introduction of the new AI Component poses challenges. Given that the AI Component relies on the implementation of all connectors in the connector repository, including it in the component repository would lead to cyclic dependencies. Therefore, the optimal solution is to consolidate all components into a single repository.

This commit

- Merges code from [connector](https://github.com/instill-ai/connector) and [operator](https://github.com/instill-ai/operator) into this repository.
- Removes unused `objectmapper`.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -42,7 +42,7 @@ There are different types of component:
 
 **Connector**
 
-- **Connectors** are used for connecting the pipeline to a vendor service. They are defined and initialized in the [connector](https://github.com/instill-ai/connector/) repository.
+- **Connectors** are used for connecting the pipeline to a vendor service. They are defined and initialized in the [connector](https://github.com/instill-ai/component/blob/main/pkg/connector).
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:45]:
```suggestion
- **Connectors** are used for connecting the pipeline to a vendor service. They are defined and initialized in the [connector](../pkg/connector) package.
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ sequenceDiagram
 
 **Operator**
 
-- An operator is used for data operations inside the pipeline. They are defined and initialized in the [operator](https://github.com/instill-ai/operator/) repository.
+- An operator is used for data operations inside the pipeline. They are defined and initialized in the [operator](https://github.com/instill-ai/component/blob/main/pkg/operator).
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:58]:
```suggestion
- An operator is used for data operations inside the pipeline. They are defined and initialized in the [operator](../pkg/operator) package.
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -122,7 +122,7 @@ When you want to contribute with a new connector or operator, you need to create
 2 configuration files define the behaviour of the component:
 
 - `definitions.json`
-    - You can refer to [OpenAI connector](https://github.com/instill-ai/connector/blob/main/pkg/openai/v0/config/definitions.json) as an example.
+    - You can refer to [OpenAI connector](https://github.com/instill-ai/component/blob/main/pkg/connector/openai/v0/config/definitions.json) as an example.
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:125]:
```suggestion
    - You can refer to [OpenAI connector](../pkg/connector/openai/v0/config/definitions.json) as an example.
```

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -134,7 +134,7 @@ When you want to contribute with a new connector or operator, you need to create
         This will showcase the upcoming component at [instill.tech](https://instill.tech).
     - We define the `resource_configuration` in this file, which defines the connector resource setup.
 - `tasks.json`
-    - You can refer to [OpenAI connector](https://github.com/instill-ai/connector/blob/main/pkg/openai/v0/config/tasks.json) as an example.
+    - You can refer to [OpenAI connector](https://github.com/instill-ai/component/blob/main/pkg/connector/openai/v0/config/tasks.json) as an example.
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:137]:
```suggestion
    - You can refer to [OpenAI connector](../pkg/connector/openai/v0/config/tasks.json) as an example.
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -175,7 +175,7 @@ type IExecution interface {
 
 <!--
 TODO:
- 1. explain how we import the connectors or operators like [here](https://github.com/instill-ai/connector/blob/main/pkg/main.go)
+ 1. explain how we import the connectors or operators like [here](https://github.com/instill-ai/component/blob/main/pkg/connector/main.go)
```
Comment (jvallesm) [inline .github/CONTRIBUTING.md:178]:
```suggestion
 1. explain how we import the connectors or operators like [here](../pkg/connector/main.go)
```

File: tools/compogen/go.sum
Comment (jvallesm) [inline tools/compogen/go.sum:1]: ❓ Why was this removed?

**Review by jvallesm — COMMENTED** [review]
Minor comments about the references to the (now local) connector and operator packages.

👍 I think this is a good move, connector and operator exist under the component umbrella and, though it makes sense to decouple the implementation of each component from the interfaces it should comply with, it doesn't justify the complexity of maintaining 3 repos.

**Comment by donch1989** [comment]
@jvallesm Thanks for the detailed review!

File: tools/compogen/go.sum
Comment (donch1989) [reply tools/compogen/go.sum:1]: Oh... Let me add it back... I might accidentally removed it.

**Review by jvallesm — APPROVED** [review]
